### PR TITLE
handle cases where SharedStorage::init isn't initializing the storage

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

**Comment by PVince81** [comment]
weird:
```
1) OCA\Provisioning_API\Tests\Controller\UsersControllerTest::testAddUserAlreadyExisting
Failed asserting that exception of type "Error" matches expected exception "OCP\AppFramework\OCS\OCSException". Message was: "Call to a member function t() on null" at
/drone/src/apps/provisioning_api/lib/Controller/UsersController.php:353
/drone/src/apps/provisioning_api/tests/Controller/UsersControllerTest.php:271
.
```

**Comment by Pytal** [comment]
> ```
> 1) OCA\Provisioning_API\Tests\Controller\UsersControllerTest::testAddUserAlreadyExisting
> Failed asserting that exception of type "Error" matches expected exception "OCP\AppFramework\OCS\OCSException". Message was: "Call to a member function t() on null" at
> /drone/src/apps/provisioning_api/lib/Controller/UsersController.php:353
> /drone/src/apps/provisioning_api/tests/Controller/UsersControllerTest.php:271
> .
> ```

Fixed on `master` in https://github.com/nextcloud/server/pull/33502/commits/4de19e9f851e2d1948d2af4dfcd70a5b22ae53cb

File: apps/files_sharing/lib/SharedStorage.php [hunk 1]
```diff
@@ -532,6 +548,17 @@ public function getSourceStorage() {
 
 	public function getWrapperStorage() {
 		$this->init();
+
+		// `init` should handle this, but apparently it sometimes doesn't
```
Comment (Pytal) [inline apps/files_sharing/lib/SharedStorage.php:552]: Is the reason for why it doesn't known?

**Comment by PVince81** [comment]
@icewind1991 is this still valid or did we fix this in a different way ? possibly through https://github.com/nextcloud/server/pull/35218

**Pull request closed without merging**